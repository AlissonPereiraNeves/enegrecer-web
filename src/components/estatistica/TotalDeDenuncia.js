import React, { Component } from 'react';
import firebaseApp from '../../utils/firebaseUtils';
import TipoDeEstatistica from './TipoDeEstatistica';

export default class TotalDeDenuncia extends Component {
  constructor(props) {
    super(props);
    this.state = { totalDeDenuncia: '0' };
  }

  componentDidMount() {
    const ref = firebaseApp.database().ref();
    ref.child('denuncias').once('value', (snapshot) => {
      this.setState({ totalDeDenuncia: snapshot.numChildren() });
    });
  }

  render() {
    return (<TipoDeEstatistica
      id="cinzaClaro"
      numero={this.state.totalDeDenuncia}
      categoria="Total de DenÃºncia"
    />
    );
  }
}
