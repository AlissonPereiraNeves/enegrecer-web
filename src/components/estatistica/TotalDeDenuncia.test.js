import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TotalDeDenuncia from './TotalDeDenuncia';

describe('Total De Denuncia', () => {
  it('deve construir a pagina corretamente ', () => {
    const totalDeDenuncia = shallow(<TotalDeDenuncia />);
    const totalDeDenunciaJson = toJson(totalDeDenuncia);
    expect(totalDeDenunciaJson).toMatchSnapshot();
  });
});
