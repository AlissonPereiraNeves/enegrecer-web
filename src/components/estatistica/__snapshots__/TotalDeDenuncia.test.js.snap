// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Total De Denuncia deve construir a pagina corretamente  1`] = `
<TipoDeEstatistica
  categoria="Total de DenÃºncia"
  id="cinzaClaro"
  numero="0"
/>
`;
